nemo/utils/exp_manager.py

```diff
@@ -158,7 +158,10 @@ def exp_manager(trainer: 'pytorch_lightning.Trainer', cfg: Optional[Union[DictCo
     if cfg.resume_if_exists:
         check_resume(trainer, log_dir, cfg.resume_past_end, cfg.resume_ignore_no_checkpoint)
 
-    cfg.name = name
+    checkpoint_name = name
+    if not checkpoint_name:  # If name returned from get_log_dir is "", use cfg.name for checkpointing
```

Comment:
Maybe `if checkpoint_name is not None or checkpoint_name != ''`?

Reply:
Wait you mean `f checkpoint_name is None or checkpoint_name == ''`, right? I'll change it to that